src/constants/wallets.ts

```diff
@@ -0,0 +1,202 @@
+import { BaseMessageSignerWalletAdapter, WalletReadyState } from '@solana/wallet-adapter-base'
+import {
+  BraveWalletAdapter,
+  Coin98WalletAdapter,
+  CoinbaseWalletAdapter, // LedgerWalletAdapter,
+  PhantomWalletAdapter,
+  SlopeWalletAdapter,
+  SolflareWalletAdapter,
+  SolletExtensionWalletAdapter,
+} from '@solana/wallet-adapter-wallets'
+import { AbstractConnector } from '@web3-react/abstract-connector'
+import { isMobile } from 'react-device-detect'
+
+import BRAVE_L from 'assets/images/brave_wallet-light.svg'
+import BRAVE from 'assets/images/brave_wallet.svg'
+import COIN98_L from 'assets/images/coin98-light.svg'
+import COIN98 from 'assets/images/coin98.svg'
+import COINBASE_L from 'assets/images/coinbase-wallet-light.svg'
+import COINBASE from 'assets/images/coinbase-wallet.svg'
+import METAMASK_L from 'assets/images/metamask-light.svg'
+import METAMASK from 'assets/images/metamask.svg'
+import PHANTOM_L from 'assets/images/phantom-light.svg'
+import PHANTOM from 'assets/images/phantom.svg'
+import SLOPE_L from 'assets/images/slope-light.svg'
+import SLOPE from 'assets/images/slope.svg'
+import SOLFLARE_L from 'assets/images/solflare-light.svg'
+import SOLFLARE from 'assets/images/solflare.svg'
+import SOLLET_L from 'assets/images/sollet-light.png'
+import SOLLET from 'assets/images/sollet.png'
+import WALLETCONNECT_L from 'assets/images/wallet-connect-light.svg'
+import WALLETCONNECT from 'assets/images/wallet-connect.svg'
+import { braveInjectedConnector, coin98InjectedConnector, injected, walletconnect, walletlink } from 'connectors'
+import checkForBraveBrowser from 'utils/checkForBraveBrowser'
+
+import { SelectedNetwork } from './networks/solana'
+
+const braveAdapter = new BraveWalletAdapter()
+const coinbaseAdapter = new CoinbaseWalletAdapter()
+const coin98Adapter = new Coin98WalletAdapter()
+const solflareAdapter = new SolflareWalletAdapter({ network: SelectedNetwork })
+const phantomAdapter = new PhantomWalletAdapter({ network: SelectedNetwork })
+const solletAdapter = new SolletExtensionWalletAdapter()
+const slopeAdapter = new SlopeWalletAdapter({ network: SelectedNetwork })
+// const ledgerAdapter = new LedgerWalletAdapter()
+
+const detectMetamask = (): WalletReadyState => {
+  if (isMobile) return WalletReadyState.Unsupported
+  // In Brave browser, by default ethereum.isMetaMask and ethereum.isBraveWallet is true even Metamask not installed
+  if (window.ethereum?.isMetaMask && !window.ethereum?.isBraveWallet) return WalletReadyState.Installed
+  return WalletReadyState.NotDetected
+}
+
+const detectBrave = (): WalletReadyState => {
+  //todo namgold: fail connect on mobile solana
```

Comment:
don't forget

Reply:
It still true, no solution yet. Let's consider it as known issue. jup.ag have same isssue too.